src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,510 @@
+use Dhii\Data\Container\CreateNotFoundExceptionCapableTrait;
+use Dhii\Data\Container\NormalizeKeyCapableTrait;
+use Dhii\Event\EventFactoryInterface;
+use Psr\Container\ContainerInterface;
+use Psr\EventManager\EventManagerInterface;
+use RebelCode\Modular\Module\AbstractBaseModule;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+
+/**
+ * Class WpBookingsUiModule.
+ *
+ * Responsible for providing UI in the dashboard.
+ *
+ * @since [*next-version*]
+ */
+class WpBookingsUiModule extends AbstractBaseModule
+{
+    /* @since [*next-version*] */
+    use ContainerGetCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateContainerExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateNotFoundExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeKeyCapableTrait;
+
+    /**
+     * Helper class to render templates using events mechanism.
+     *
+     * @var TemplateManager
+     */
+    protected $templateManager;
+
+    /**
+     * Registered booking's page ID.
+     *
+     * @var string
+     */
+    protected $bookingsPageId;
+
+    /**
+     * Page where metabox application should be shown.
+     *
+     * @var string
+     */
+    protected $metaboxPageId;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable         $key                  The module key.
+     * @param string[]|Stringable[]     $dependencies         The module  dependencies.
+     * @param ContainerFactoryInterface $configFactory        The config factory.
+     * @param ContainerFactoryInterface $containerFactory     The container factory.
+     * @param ContainerFactoryInterface $compContainerFactory The composite container factory.
+     * @param EventManagerInterface     $eventManager         The event manager.
+     * @param EventFactoryInterface     $eventFactory         The event factory.
+     */
+    public function __construct(
+        $key,
+        $dependencies,
+        $configFactory,
+        $containerFactory,
+        $compContainerFactory,
+        $eventManager,
+        $eventFactory
+    ) {
+        $this->_initModule($key, $dependencies, $configFactory, $containerFactory, $compContainerFactory);
+        $this->_initModuleEvents($eventManager, $eventFactory);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function setup()
+    {
+        return $this->_setupContainer(
+            $this->_loadPhpConfigFile(WP_BOOKINGS_UI_MODULE_CONFIG_FILE),
+            $this->_getServicesDefinitions()
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function run(ContainerInterface $c = null)
+    {
+        $this->metaboxPageId   = $c->get('wp_bookings_ui/metabox/post_type');
+        $this->templateManager = $c->get('template_manager');
```

Comment:
Template manager is unused in this scope. Also, not sure why you would need to store it in a property.

Reply:
I've stored it in the property because I use it in the method for rendering.

Reply:
But do you really need to access it this way? A better way would be to abstract the retrieval of the template manager, and get it from the container.